simulate/simulate.cc

```diff
@@ -0,0 +1,1846 @@
+// Copyright 2021 DeepMind Technologies Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "simulate.h"
+
+#include <chrono>
+#include <cstdio>
+#include <cstring>
+#include <mutex>
+#include <string>
+#include <thread>
+
+#include <mjxmacro.h>
+#include "array_safety.h"
+namespace mju = ::mujoco::sample_util;
+
+
+//-------------------------------- global -----------------------------------------------
+
+const int maxgeom = 5000;           // preallocated geom array in mjvScene
```

Comment:
This (and other constants, and all local functions) should be declared `static` or placed in an unnamed-namespace.